File: LeetCode/LengthOfLongestSubstring.py
class Solution:
    # @return an integer
    def lengthOfLongestSubstring(self, s):
        l = r = ans = 0
        n = len(s)
        occ = set()
        while r < n:
            while s[r] in occ:
                occ.remove(s[l])
                l += 1
            occ.add(s[r])
            r += 1
            ans = max(ans, r - l)
        return ans

File: LeetCode/LengthOfLongestSubstring.cc
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        // IMPORTANT: Please reset any member data you declared, as
        // the same Solution instance will be reused for each test case.
        bool occur[256];
        memset(occur, false, sizeof(occur));
        auto left = s.begin(), right = s.begin();
        int ans = 0;
        while(right != s.end()) {
            while(occur[*right])
                occur[*(left++)] = false;
            occur[*(right++)] = true;
            ans = max(ans, right - left);
        }
        return ans;
    }
};